undertow/WFLY-11031_Formatted_Access_Logs.adoc

```diff
@@ -0,0 +1,71 @@
+= JSON Formatted Access Logs
+:author:            James Perkins
+:email:             jperkins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently access logs can only be written as plain text. This plain text will be written to either a specific file or
+logged via the normal logging facilities. It's desirable for a log aggregator to accept formatted log messages. Using
+JSON seems to be the most common format.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11031[WFLY-11031]
+* https://issues.jboss.org/browse/EAP7-1183[EAP7-1183]
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Affected Projects or Components
+
+* WildFly
+** Undertow subsystem
+
+=== Other Interested Projects
+
+* OpenShift
+
+== Requirements
+
+=== Hard Requirements
+
+* The output must be well formed JSON.
+* The formatted message must be on a single line.
+* The output must be able to write to standard out. This is required for OpenShift.
+* A label of some sort, whether it's hard-coded or user defined, must be added to the output. This allows filtering
+  log messages of different types.
+
+=== Nice-to-Have Requirements
+
+* Allow the user to choose which attributes are added to the JSON.
+* The ability to override the key name.
+* The ability to add custom key/value pairs to the output.
+    ** This would be something like `{ "@version" : "1" }`
+* We could consider allowing a pretty-print version of the data as well.
+
+=== Non-Requirements
+
+== Implementation Plan
+
+The implementation will be done in WildFly and we should not need to do anything to Undertow itself.
+
+This is not a complete implementation plan. However it might be desirable to deprecate the current
+`/subsystem=undertow/server=default-server/host=default-host/setting=access-log`. Adding new attributes for a JSON
+formatter and the ability to write to standard out might make the model confusing. This is yet to be determined though.
```

Comment:
Has this been determined somehow already? Could you please describe how the configuration may look like?

Reply:
Will do. I've got it working locally so I can put down what I did which will hopefully explain it better.